### ISSUE TITLE
Negative scaling of inner() atom

### ISSUE THREAD

**Issue by rjboczar** [cvxgrp/dsp#5]
Not sure I'm using it correctly, but here is a MWE:

```
import dsp
import cvxpy as cvx
from numpy.random import default_rng

rng = default_rng(seed=777)


def mwe():
    n = 5
    A = rng.integers(low=1, high=5, size=(n, n))
    print(A)
    x = cvx.Variable(n)
    y = cvx.Variable(n)
    constraints = [x >= 0, y >= 0, cvx.sum(x) == 1, cvx.sum(y) == 1]

    objs = [
        dsp.MinimizeMaximize(f)
        for f in (
            dsp.inner(-x, A @ y),
            dsp.inner(x, -A @ y),
            -dsp.inner(x, A @ y),
        )
    ]

    for obj in objs:
        prob = dsp.SaddlePointProblem(obj, constraints)
        prob.solve()
        print(prob.value)


if __name__ == "__main__":
    mwe()
```

This prints out
```
[[4 3 2 2 1]
 [3 2 4 2 1]
 [2 2 2 3 2]
 [3 2 3 1 2]
 [4 1 3 1 3]]
-2.00000000020348
-2.00000000020348
-2.333333332964706
```

Why is the third one different, and which is the right way? The docs just reference scaling in general, not restricting it to positive.

**Comment by phschiele** [comment]
@rjboczar Thanks for the reproducible report, we should improve the documentation for negation, as there are some subtleties to it.

In your example `A`, `x`, and `y` are nonnegative. In the first two cases, the minus is inside the `inner` expression, i.e., the expression is nonpositive. Recall that the expression is minimized in the first and maximized in the second argument. Loosely speaking, that means that `x` wants to "choose" large values in `A` and `y` wants to choose small values.
This is $\min_x \max_y -x^TAy$.

In the third case, the expression does not have a minus inside and is therefore nonnegative. Now `x` wants to choose small values in `A` and `y` wants to choose large values. The result is then negated, i.e., $-(\min_x \max_y x^TAy)$, which ist the same as $\min_y \max_x -x^TAy$ (notice the change in subscripts).

You can see the roles of the variables in the problem by printing `prob.convex_variables()` and `prob.concave_variables()`. 

Compare section 3.3 of [Juditsky and Nemirovski](https://arxiv.org/pdf/2102.01002.pdf) for more details.

**Comment by rjboczar** [comment]
So you are saying e.g. `dsp.MinimizeMaximize(-dsp.inner(x,y))` actually returns 

$-\min_x\max_y x^\top y$ = $\max_x\min_y -x^\top y \neq \min_x\max_y - x^\top y$

?

**Comment by phschiele** [comment]
@rjboczar exactly. 
Here is an even simpler example:
```py
import cvxpy as cp
import dsp

x = cp.Variable(name='x')
y = cp.Variable(name='y')
constraints = [1 <= x, x <= 2, 3 <= y, y <= 4]

prob1 = dsp.SaddlePointProblem(dsp.MinimizeMaximize(dsp.inner(-x,y)), constraints)
prob2 = dsp.SaddlePointProblem(dsp.MinimizeMaximize(-dsp.inner(y,x)), constraints)

prob1.solve()
>>> -6.0
prob2.solve()
>>> -6.0

print(prob1.x_prob)  # The dualized problem
>>> minimize reshape(-x, (1,), F) @ [0.] + reshape(lamb, (2,), F) @ [-3.  4.]
subject to 1.0 <= x
           x <= 2.0
             (0, 0)	-1.0
  (0, 1)	1.0 @ reshape(lamb, (2,), F) ==   (0, 0)	1.0 @ reshape(-x, (1,), F)
           0.0 <= reshape(lamb, (2,), F)[0:2]

print(prob2.x_prob)  # The dualized problem
>>> minimize reshape(-x, (1,), F) @ [0.] + reshape(lamb, (2,), F) @ [-3.  4.]
subject to 1.0 <= x
           x <= 2.0
             (0, 0)	-1.0
  (0, 1)	1.0 @ reshape(lamb, (2,), F) ==   (0, 0)	1.0 @ reshape(-x, (1,), F)
           0.0 <= reshape(lamb, (2,), F)[0:2]
```

**Comment by rjboczar** [comment]
Gotcha, thanks. Not sure if the fix is in code or documentation, but the fact that `dsp.inner(-x,y)` is not the same as `-dsp.inner(x,y)` feels like a bit of a gotcha.

**Comment by phschiele** [comment]
Addressed in [aaf5f92](https://github.com/cvxgrp/dsp/commit/aaf5f926eed6c1f5bff6cf111e9ad1f946975a16).
